enhancements/implicit-catalog-versioning.md

```diff
@@ -0,0 +1,152 @@
+- Add a required `--mode` flag to `opm index` and `opm bundle` commands that can determine how fine grained the update is inserted into each channel's graph
+
+### Ignore the `replaces` field when building the registry graph, instead use `version`:
+
+Currently, the update graph defined when creating an operator catalog is explicit. Essentially, the ClusterServiceVersion (CSV) file has an explicit `replaces` field:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: etcdoperator.v0.9.2
+  namespace: placeholder
+spec:
+    displayName: etcd
+    description: Etcd Operator
+    version: 0.9.2
+    replaces: etcdoperator.v0.9.0
+```
+
+The `replaces` field is the source for the update graph for each install channel in a given operator. When the `operator-registry` builds the update graph, it actually walks through each CSV in a given Operator and, for each channel, constructs a chain of CSVs that define the update path. Instead, we can just define our spec without the replaces field:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: etcdoperator.v0.9.2
+  namespace: placeholder
+spec:
+    displayName: etcd
+    description: Etcd Operator
+    version: 0.9.2
+```
+
+We would use the fact that 0.9.2 > 0.9.0 to implicitly assume that 0.9.0 is replaced by 0.9.2 through the rules of semantic version.
+
+To accomplish this:
+- Make `spec.version` a required field in the operator-registry project
+- Update the `operator-registry` to walk through the set of CSVs and construct the replacement chain by reading the `version` field instead of `replaces`. This will be done by modifying the Channel tables to point to next ordered semantic versions when defining the graph
+- Ignore `skips` field entirely when building the graph
+- Update OLM to use the query for the replacement rather than the current value in the existing CSV
+
+Note that these requirements will only be the case when generating update graphs with the latest version of the operator-registry's `opm registry` commands. 
+
+### Mode
+
+First, a quick note about `skipRange`: The `skipRange` annotation was originally added to allow an explicit definition of version ranges that can be skipped over -- generally this was, done in order to allow patch versions of released operators to upgrade directly. However, given that version is becoming a first class citizen, some users may expect that patch version to work in that fashion in general.
+
+In order to account for this possibility, we will explicitly ask the user when adding bundles which method they want to use to upgrade. To handle this, `opm index` and `opm bundle` will add a new flag `--mode` that enumerates to three different options:
+
+`opm index add quay.io/namespace/my-operator --mode=semver`
```

Comment:
This doesn't change the connectedness of the entire index, just the given node, correct?

Reply:
This will modify the database so that when OLM queries for the next item in the replacement chain it can respond accordingly for that specific version. If we want to update it for all nodes in a given index or a given package we would need to provide a separate command for that.